### PR TITLE
Changed the type of automatic field and added support for customer_managed_encryption in automatic field

### PR DESCRIPTION
**Pull request by abheda-crest**
Changed the type of the `automatic` field from TypeBool to TypeList and added support for the `replication.automatic.customer_managed_encryption` field in the `google_secret_manager_secret` resource.

fixes https://github.com/hashicorp/terraform-provider-google/issues/10679
fixes https://github.com/hashicorp/terraform-provider-google/issues/9272
fixes https://github.com/hashicorp/terraform-provider-google/issues/14282

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
secretmanager: changed the type of `automatic` field from TypeBool to TypeList
```
```release-note:enhancement
secretmanager: added `replication.automatic.customer_managed_encryption` field to `google_secret_manager_secret` resource
```

### CODE REVIEW

**Comment by abheda-crest** [comment]
@slevenick While generating the providers after changing the automatic field to block type in the terraform configs (i.e tests and examples), I encountered that for the documentation for the resources [`google_dataform_repository`](https://github.com/hashicorp/terraform-provider-google/blob/main/website/docs/r/dataform_repository.html.markdown?plain=1#L52) and [`google_dataform_repository_release_config`](https://github.com/hashicorp/terraform-provider-google/blob/main/website/docs/r/dataform_repository_release_config.html.markdown?plain=1#L52), the automatic field remains boolean i.e `automatic = true` even though I've changed them in the examples [`dataform_repository.tf.erb`](https://github.com/GoogleCloudPlatform/magic-modules/blob/cc8bee13549be0743c8d14b71cd287d353ce993e/mmv1/templates/terraform/examples/dataform_repository.tf.erb#L11) and [`dataform_repository_release_config.tf.erb`](https://github.com/GoogleCloudPlatform/magic-modules/blob/cc8bee13549be0743c8d14b71cd287d353ce993e/mmv1/templates/terraform/examples/dataform_repository_release_config.tf.erb#L11) respectively. I am facing this issue while generating the GA provider and not the Beta provider. I suspect this is something related to the documentation being generated for the GA provider even though these resources are [beta only](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataform_repository).

**Comment by abheda-crest** [comment]
@slevenick Could you please review this PR?

**Comment by slevenick** [comment]
I need to check if this is an acceptable change.

My understanding is that we need to support an upgrade path through the last 4.x version to 5.0.0, which may not be possible with this PR.

File: mmv1/templates/terraform/examples/secret_with_automatic_cmek.tf.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  member        = "serviceAccount:service-${data.google_project.project.number}@gcp-sa-secretmanager.iam.gserviceaccount.com"
+}
+
+resource "google_secret_manager_secret" "<%= ctx[:primary_resource_id] %>" {
+  secret_id = "<%= ctx[:vars]['secret_id'] %>"
+
+  replication {
+    automatic {
+      customer_managed_encryption {
+        kms_key_name = "<%= ctx[:vars]['kms_key_name'] %>"
```
Comment (gptSanyam) [inline mmv1/templates/terraform/examples/secret_with_automatic_cmek.tf.erb:15]: Where is the kms_key and keyring defined?
For automatic replication location is global and CMEK should also be defined in a global keyring.

Comment (abheda-crest) [reply mmv1/templates/terraform/examples/secret_with_automatic_cmek.tf.erb:15]: I've used the function `BootstrapKMSKey` as mentioned in the [Secret.yaml](https://github.com/GoogleCloudPlatform/magic-modules/pull/8662/files#diff-9001ca5000d45e69e6d14ce2f520e2e82e8a77c74fde2affd94bafdc8dd29db3R50) that will use a particular keyring and a particular key i.e. `tftest-shared-keyring-1` and `tftest-shared-key-1` respectively in the test environment. The implementation of this function can be found here: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/acctest/bootstrap_test_utils.go#L42.

**Comment by abheda-crest** [comment]
> See #8721
> 
> I don't think this is an acceptable change to make as implemented here.
> 
> The alternative that I can see would be changing the name of the field so that we can support the side-by-side of the old field and the new field in the same version of 4.x, and then removing the old field in 5.0.0

Just FYI, the `automatic` field is showing ForceNew behavior and the new field to be added must also be kept ForceNew as patching the `replication` field from `automatic` to `user_managed` (and vice-versa) throws an error 400 `Existing secret has automatic replication, but requested secret does not. Updating secret replication is not supported.` Thus, Migrating from `automatic` field to the new field will recreate the secret.

**Comment by slevenick** [comment]
> > See #8721
> > I don't think this is an acceptable change to make as implemented here.
> > The alternative that I can see would be changing the name of the field so that we can support the side-by-side of the old field and the new field in the same version of 4.x, and then removing the old field in 5.0.0
> 
> @slevenick Just FYI, the `automatic` field is showing ForceNew behavior and the new field to be added must also be kept ForceNew as patching the `replication` field from `automatic` to `user_managed` (and vice-versa) throws an error 400 `Existing secret has automatic replication, but requested secret does not. Updating secret replication is not supported.` Thus, Migrating from `automatic` field to the new field will recreate the secret.

It's definitely hard to do to support two fields side-by-side, but I think it can be done. It might require a complicated CustomizeDiff function in the version where both are supported

**Comment by abheda-crest** [comment]
> > > See #8721
> > > I don't think this is an acceptable change to make as implemented here.
> > > The alternative that I can see would be changing the name of the field so that we can support the side-by-side of the old field and the new field in the same version of 4.x, and then removing the old field in 5.0.0
> > 
> > 
> > @slevenick Just FYI, the `automatic` field is showing ForceNew behavior and the new field to be added must also be kept ForceNew as patching the `replication` field from `automatic` to `user_managed` (and vice-versa) throws an error 400 `Existing secret has automatic replication, but requested secret does not. Updating secret replication is not supported.` Thus, Migrating from `automatic` field to the new field will recreate the secret.
> 
> It's definitely hard to do to support two fields side-by-side, but I think it can be done. It might require a complicated CustomizeDiff function in the version where both are supported

@slevenick I am thinking of adding custom_flatten and custom_expand for the `replication` field to check which of the two fields are provided and to call the respective flatteners and expanders.

**Comment by c2thorn** [comment]
This was abandoned in favor of https://github.com/GoogleCloudPlatform/magic-modules/pull/8838

**Pull request closed without merging**